source_code.py

```
@@ -110,43 +85,69 @@ def query_a(results_queue):
+def check_mx_reputation(domain):
+    """Return 'Bad' if any MX IP is listed in DNSBL, else 'Normal'"""
+    try:
+        mx_records = dns.resolver.resolve(domain, 'MX')
+        for mx in mx_records:
+            host = str(mx.exchange).rstrip('.')
+            ips = dns.resolver.resolve(host, 'A')
+            for ip in ips:
+                if lookup_dnsbl(ip.to_text()):
+                    return 'Bad'
+    except Exception:
+        pass
```

Ruff (0.12.2)

`try`-`except`-`pass` detected, consider logging the exception

(S110)